zulipterminal/ui_tools/views.py

```diff
@@ -1569,6 +1570,65 @@ def keypress(self, size: urwid_Size, key: str) -> str:
         return super().keypress(size, key)
 
 
+class TopicInfoView(PopUpView):
+    def __init__(self, controller: Any, stream_id: int, topic: str) -> None:
+        self.stream_id = stream_id
+        self.controller = controller
+        stream = controller.model.stream_dict[stream_id]
+        self.topic_name = topic
+        stream_name = stream["name"]
+
+        title = f"{stream_name} {STREAM_TOPIC_SEPARATOR} {self.topic_name}"
+
+        topic_info_content: PopUpViewTableContent = []
+
+        popup_width, column_widths = self.calculate_table_widths(
+            topic_info_content, len(title)
+        )
+
+        if self.topic_name.startswith(RESOLVED_TOPIC_PREFIX):
+            self.resolve_topic_setting_btn_lbl = "Unresolve Topic"
+        else:
+            self.resolve_topic_setting_btn_lbl = "Resolve Topic"
+        resolve_topic_setting = urwid.Button(
+            self.resolve_topic_setting_btn_lbl,
+            self.toggle_resolve_status,
+        )
+
+        curs_pos = len(self.resolve_topic_setting_btn_lbl) + 1
+        # This shifts the cursor present over the first character of
+        # resolve_topic_button_setting label to last character + 1 so that it isn't
+        # visible
+
+        resolve_topic_setting._w = urwid.AttrMap(
+            urwid.SelectableIcon(
+                self.resolve_topic_setting_btn_lbl, cursor_position=curs_pos
+            ),
+            None,
+            "selected",
+        )
+
+        # Manual because calculate_table_widths does not support buttons.
+        # Add 4 to button label to accommodate the buttons itself.
+        popup_width = max(
+            popup_width,
+            len(resolve_topic_setting.label) + 4,
+        )
+
+        self.widgets = self.make_table_with_categories(
+            topic_info_content, column_widths
+        )
+
+        self.widgets.append(resolve_topic_setting)
+        super().__init__(controller, self.widgets, "TOPIC_INFO", popup_width, title)
```

Comment:
This TOPIC_INFO has no meaning here yet. Not sure of the ordering; let's leave it for now.